app/src/main/java/com/platypus/android/server/VehicleServerImpl.java

```diff
@@ -479,15 +489,23 @@ protected void onCommand(JSONObject cmd) {
 
                             // TODO: Remove this hack to store winch depth
                             winch_depth_ = reading.data[0];
+                        } else {
+                            Log.w(TAG, "Received sensor data of unknown type '" + cmd + "'.");
+                            mLogger.warn(cmd);
+
+                            // Don't want to call sendSensor on unknown sensor type
+                            continue;
                         }
 
-                        // Send out and log the collected sensor reading.
-                        sendSensor(sensor, reading);
+                        // Attempt to log the collected sensor reading
                         mLogger.info(new JSONObject()
                                 .put("sensor", new JSONObject()
                                         .put("channel", reading.channel)
                                         .put("type", reading.type.toString())
                                         .put("data", new JSONArray(reading.data))));
+
+                        // Send out the collected sensor reading
+                        sendSensor(sensor, reading);
```

Comment:
just reordered the sending to be after logging so if there are further crashes we have the string logged.